g1ant-language/desktop-apps-automation2.md

```diff
@@ -0,0 +1,211 @@
+
+### Reading Emails
+
+The whole process described in your plan of actions can be divided into three main tasks: retrieving emails, searching for specific messages and deleting email addresses from the list. Let’s make procedures out of these tasks and name them `checkmail`, `checksubject` and `deleteaddress`.
+
+```G1ANT
+procedure checkmail
+    
+end procedure
+
+procedure checksubject
+    
+end procedure
+
+procedure deleteaddress
+    
+end procedure
+```
+
+Start with the `checkmail` procedure. It should simply login to your email account and retrieve messages. The `mail.imap` command does it in one fell swoop. What it needs is the email server information, your login credentials and a name of the variable, which will store emails. You can also specify some filtering, which is helpful when processing huge mailboxes: time range (start and end dates of messages), an option to process only unread messages and whether to mark them as read afterwards. The full syntax for the command is as follows:
+
+```G1ANT
+mail.imap host ‴imap.server.name‴ port ‴993‴ login ‴your@email.address‴ password ‴p@$$w0rd‴ sincedate ♥startdate todate ♥enddate onlyunreadmessages true markallmessagesasread true result ♥result
+```
+
+Let’s assume your script should check only unread messages received since the day before the robot is run. These messages will be marked as read (it’s the default value, so you don’t have to specify it) and will be passed to the `♥emails` variable. So here is your `checkmail` procedure:
+
+```G1ANT
+procedure checkmail
+	mail.imap host ‴imap.server.name‴ port ‴993‴ login ‴your@email.address‴ password ‴p@$$w0rd‴ sincedate ♥yesterday onlyunreadmessages true result ♥emails
+end procedure
+```
+
+Of course, you should provide your true server name, login and password in the code above.
+
+There are two variables used in this procedure: `♥yesterday` and `♥emails`. The latter is created automatically by the command (the default name is `♥result`), but the first needs to be declared beforehand. Let’s do it.
+
+### Injecting C# Code
+
+It’s easy to get the current date in G1ANT.Language — all you need is to use the `♥date` special variable. But    things get tough when you want to make some operations on a date, for example to add or subtract days. Sometimes it’s better not to invent a wheel, rather to use existing functions from C# language instead. These C# “injections” — called snippets — into G1ANT.Language are done using `⊂⊃` special characters (`Insert/Macro` menu, **Ctrl+9** keyboard shortcut or the `⊂⊃` icon on the toolbar).
```

Comment:
If I was a C# programmer, I probably would use "methods", but it's totally unclear to people like me. Average user doesn't understand "methods", contrary to "functions".

Reply:
I would like to know what our team thinks about it, maybe they would agree with you. I believe it is not good to use incorrect names just because it MIGHT be more easy to understand for others. I believe our manual should be as professional as it can and I find it unprofessional because C# has methods, not functions.

Reply:
let's meet halfway and change to this: "functions (methods, to be precise)"

Reply:
```suggestion
It’s easy to get the current date in G1ANT.Language — all you need is to use the `♥date` special variable. But    things get tough when you want to make some operations on a date, for example to add or subtract days. Sometimes it’s better not to invent a wheel, rather to use existing functions (methods, to be precise) from C# language instead. These C# “injections” — called snippets — into G1ANT.Language are done using `⊂⊃` special characters (`Insert/Macro` menu, **Ctrl+9** keyboard shortcut or the `⊂⊃` icon on the toolbar).
```

Reply:
ok